shared/utility/math/geometry/Quad.hpp

```diff
@@ -32,199 +31,6 @@ namespace math {
-                T diag1 = bl - tr;
-                T diag2 = tl - br;
-                return std::abs(0.5 * ((diag1(0) * diag2(1)) - (diag1(1) * diag2(0))));
-            }
-            Scalar aspectRatio() const {
-                return ((arma::norm(br - tr) + arma::norm(bl - tl) + 2)
-                        / (arma::norm(br - bl) + arma::norm(tr - tl) + 2));
-            }
-
-            std::vector<T> getVertices() const {
-                std::vector<T> vert = {tr, br, bl, tl};
-                return vert;
-            }
-
-            bool overlapsHorizontally(const Quad& other) const {
-                // Rough for now.
-                Scalar farRight   = std::max(tr(0), br(0));
-                Scalar farLeft    = std::min(tl(0), bl(0));
-                Scalar o_farRight = std::max(other.tr(0), other.br(0));
-                Scalar o_farLeft  = std::min(other.tl(0), other.bl(0));
-
-                return !((farRight < o_farLeft) || (o_farRight < farLeft));
-            }
-
-            bool checkCornersValid() const {
-                return br.n_elem == 2 && bl.n_elem == 2 && tr.n_elem == 2 && tl.n_elem == 2;
-            }
-
-            static Quad getBoundingBox(const std::vector<T>& points) {
-                // Check for
-                if (points.size() <= 0) {
-                    throw std::domain_error("Request made for bounding box for empty list of points!");
-                }
-
-                Scalar min_x = points[0](0);
-                Scalar max_x = points[0](0);
-                Scalar min_y = points[0](1);
-                Scalar max_y = points[0](1);
-                for (uint i = 1; i < points.size(); i++) {
-                    auto& p = points[i];
-                    max_x   = std::max(max_x, p(0));
-                    min_x   = std::min(min_x, p(0));
-                    max_y   = std::max(max_y, p(1));
-                    min_y   = std::min(min_y, p(1));
-                }
-                return Quad(T({min_x, min_y}), T({min_x, max_y}), T({max_x, max_y}), T({max_x, min_y}));
-            }
-
-        private:
-            T bl;  //! @variable The bottom-left of the Quad.
-            T br;  //! @variable The bottom-right of the Quad.
-            T tr;  //! @variable The top-right of the Quad.
-            T tl;  //! @variable The top-left of the Quad.
-
-            // //! @brief output stream operator.
-            // template <typename U>
-            // friend std::ostream& operator<<(std::ostream& output, const Quad<U>& quad);
-
-            // //! @brief output stream operator for a vector of goals.
-            // template <typename U>
-            // friend std::ostream& operator<<(std::ostream& output, const std::vector<Quad<U>>& quads);
-        };
-
-        // *****************
-        // * EIGEN VERSION *
-        // *****************
-
         template <typename Scalar, int R, int C>
         class Quad<typename Eigen::Matrix<Scalar, R, C>> {
```

Comment:
We should be able to remove template specialisation from this class now
```suggestion
        class Quad {
```

Reply:
I removed it, but I don't really think it made it more readable in the end. I'll let you decide I think.